### PR TITLE
Added nutrition subtopic to allow nutrition fact-related queries

### PR DESCRIPTION
**Pull request by paulgray01**
added nutrition values to mock product data and added match pattern and message handler for nutrition queries

### CODE REVIEW

**Review by SanaBob — APPROVED** [review]
lgtm

**Comment by thuanGIT** [comment]
New PR on `product_store_query`

**Comment by SanaBob** [comment]
added unit test
fixed database

File: app/products/database.py [hunk 1]
```diff
@@ -64,7 +64,12 @@ def init_database(self):
                 names VARCHAR(55) NOT NULL,
                 price DECIMAL(8,2),
                 price_scale VARCHAR(10),
-                in_stock INT NOT NULL
+                in_stock INT NOT NULL,
+                calories INT NOT NULL,
```
Comment (thuanGIT) [inline app/products/database.py:68]: There is no need for nutrition facts to be NOT NULL as they might be missing.

File: app/products/product_info.py [hunk 2]
```diff
@@ -16,6 +16,8 @@ def create_match_paterns(self):
         self.price_pattern = re.compile(
             r"(price|cost|how much)", re.IGNORECASE)
         self.stock_pattern = re.compile(r"(stock|how many)", re.IGNORECASE)
+        self.nutrition_pattern = re.compile(
+            r"(calories|protein|carbs|carbohydrates|sugar|fat|nutrition|nutritional)", re.IGNORECASE)
```
Comment (thuanGIT) [inline app/products/product_info.py:20]: A nice to have is to handle questions like "Does this bread make me gain weight?"

File: test/test_product_info.py [hunk 3]
```diff
@@ -30,4 +30,32 @@ def test_handler_stockpear(self, classTest):
         expectedOutput = "{} are out of stock.".format(MOCK_PRODUCT_DATA[3]["names"].capitalize())
         assert(classTest.handle(message) == expectedOutput)
 
+@pytest.mark.prod_info
+class TestNutritionInfo:
```
Comment (thuanGIT) [inline test/test_product_info.py:34]: Should add a case where the handler fails to ensure it does not give answers for nothing.

Comment (thuanGIT) [reply test/test_product_info.py:34]: @SanaBob

**Comment by thuanGIT** [comment]
@paulgray01 @SanaBob The other team wants to talk to you to migrate to use DialogFlow for NLP instead of regex. So we will have this PR as pending until the discussion is resolved.

**Comment by SanaBob** [comment]
Added endpoint for Diagflow integration

File: app/products/product_info.py [hunk 4]
```diff
@@ -23,6 +23,8 @@ def dispose(self):
         super().dispose()
 
     def handle_prod_intent(self, product: str, intent: str) -> str:
+        
+        intent = intent[8:] # hardcoded to filter intent: product-<intent> Ex. product-price -> intent = price
```
Comment (thuanGIT) [inline app/products/product_info.py:27]: You can use String.index() or String.split() for more flexible handling.

File: app/products/product_info.py [hunk 5]
```diff
@@ -22,7 +22,24 @@ def create_match_paterns(self):
     def dispose(self):
         super().dispose()
 
-    def handle(self, message: str) -> str:
+    def handle_prod_intent(self, product: str, intent: str) -> str:
+
+        request = None
+
+        for prod in MOCK_PRODUCT_DATA:
```
Comment (thuanGIT) [inline app/products/product_info.py:31]: Please use `execute_query()` to search for product. This method does a linear scan, which is very expensive,

**Review by thuanGIT — CHANGES_REQUESTED** [review]
With new implementation with Dialogflow, we need to redesign our internal bot framework. The current new method names can be quite easy to be misunderstood.

We will resolve this in our next meeting.

**Review by thuanGIT — APPROVED** [review]

**Pull request merged**